Tophat/Utilities/ArtifactDownloader.swift

```diff
@@ -46,11 +46,18 @@ final class ArtifactDownloader: NSObject {
 		do {
 			defer { progressObservation = nil }
 
-			let (localURL, _) = try await URLSession.shared.download(from: url, delegate: self)
+			let (localURL, response) = try await URLSession.shared.download(from: url, delegate: self)
+			let statusCode = (response as? HTTPURLResponse)?.statusCode ?? 0
+			if statusCode != 200 {
```

Comment:
Does this assumption stand for URLs that redirect?  I haven't encountered a redirecting URL for any artifacts I've worked with, but being mindful of others that may be relying on this today.

Also, could potentially turn this into a `guard`.